src/analysis/analytic/basic.lean

```diff
@@ -448,3 +453,315 @@ begin
+    by { ext b, rcases b with ⟨n, s⟩, simp [Bnnnorm, nnreal.coe_pow, coe_nnnorm] },
+  rw [this, nnreal.summable_coe, ← ennreal.tsum_coe_ne_top_iff_summable],
+  apply ne_of_lt,
+  calc (∑ b, ↑(Bnnnorm b))
+  = (∑ n, (∑ s, ↑(Bnnnorm ⟨n, s⟩))) : by exact ennreal.tsum_sigma' _
+  ... ≤ (∑ n, (((nnnorm (p n) * (nnnorm x + r)^n) : nnreal) : ennreal)) :
+    begin
+      refine ennreal.tsum_le_tsum (λ n, _),
+      rw [tsum_fintype, ← ennreal.coe_finset_sum, ennreal.coe_le_coe],
+      apply le_of_eq,
+      calc finset.univ.sum (λ (s : finset (fin n)), Bnnnorm ⟨n, s⟩)
+      = finset.univ.sum (λ (s : finset (fin n)),
+      nnnorm (p n) * ((nnnorm x) ^ (n - s.card) * r ^ s.card)) :
+        by simp [← mul_assoc]
+      ... = nnnorm (p n) * (nnnorm x + r) ^ n :
+      by { rw [add_comm, ← finset.mul_sum, ← fin.sum_pow_mul_eq_add_pow], congr, ext s, ring }
+    end
+  ... ≤ (∑ (n : ℕ), (C * a ^ n : ennreal)) :
+    tsum_le_tsum (λ n, by exact_mod_cast hC n) ennreal.summable ennreal.summable
+  ... < ⊤ :
+    by simp [ennreal.mul_eq_top, ha, ennreal.tsum_mul_left, ennreal.tsum_geometric,
+              ennreal.lt_top_iff_ne_top]
+end
+
+/-- Auxiliary lemma controlling the summability of the sequence appearing in the definition of
+`p.change_origin`, second version. -/
+lemma change_origin_summable_aux2 (h : (nnnorm x + r : ennreal) < p.radius) :
+  @summable ℝ _ _ _ ((λ ⟨k, n, s, hs⟩, ∥(p n).restr s hs x∥ * ↑r ^ k) :
+    (Σ (k : ℕ) (n : ℕ), {s : finset (fin n) // finset.card s = k}) → ℝ) :=
+begin
+  let γ : ℕ → Type* := λ k, (Σ (n : ℕ), {s : finset (fin n) // s.card = k}),
+  let Bnorm : (Σ (n : ℕ), finset (fin n)) → ℝ := λ ⟨n, s⟩, ∥p n∥ * ∥x∥ ^ (n - s.card) * r ^ s.card,
+  have SBnorm : summable Bnorm := p.change_origin_summable_aux1 h,
+  let Anorm : (Σ (n : ℕ), finset (fin n)) → ℝ := λ ⟨n, s⟩, ∥(p n).restr s rfl x∥ * r ^ s.card,
+  have SAnorm : summable Anorm,
+  { refine summable_of_norm_bounded _ SBnorm (λ i, _),
+    rcases i with ⟨n, s⟩,
+    suffices H : ∥(p n).restr s rfl x∥ * (r : ℝ) ^ s.card ≤
+      (∥p n∥ * ∥x∥ ^ (n - finset.card s) * r ^ s.card),
+    { have : ∥(r: ℝ)∥ = r, by rw [real.norm_eq_abs, abs_of_nonneg (nnreal.coe_nonneg _)],
+      simpa [Anorm, Bnorm, this] using H },
+    exact mul_le_mul_of_nonneg_right ((p n).norm_restr s rfl x)
+      (pow_nonneg (nnreal.coe_nonneg _) _) },
+  let e : (Σ (n : ℕ), finset (fin n)) ≃
+      (Σ (k : ℕ) (n : ℕ), {s : finset (fin n) // finset.card s = k}) :=
+    { to_fun := λ ⟨n, s⟩, ⟨s.card, n, s, rfl⟩,
+      inv_fun := λ ⟨k, n, s, hs⟩, ⟨n, s⟩,
+      left_inv := λ ⟨n, s⟩, rfl,
+      right_inv := λ ⟨k, n, s, hs⟩, by { induction hs, refl } },
+  rw ← e.summable_iff,
+  convert SAnorm,
+  ext i,
+  rcases i with ⟨n, s⟩,
+  refl
+end
+
+/-- Auxiliary lemma controlling the summability of the sequence appearing in the definition of
+`p.change_origin`, third version. -/
+lemma change_origin_summable_aux3 (k : ℕ) (h : (nnnorm x : ennreal) < p.radius) :
```

Comment:
This lemma causes a deterministic timeout with `-T50000`. Can you slice it any thinner? ("No" is a reasonable answer; maybe we should just pick a different value of 50000 going forward.)

Reply:
The reason is that I use lambdas with brackets, which means there is some trivial clean-up to be done at several places, and I leave it to `tidy`. I can try to optimize the output of `tidy?` to see if I can make it more efficient. And the answer to your question (can I slice it?) is no, it is already essentially maximally sliced.

Reply:
In that case my vote is that we point to examples like this when we say it's now the `-T100000` challenge. :-)